eval/api_runner.py

```diff
@@ -14,30 +14,61 @@
 from utils.reporting import upload_results
 
 
-def process_row(row, api_url, num_beams, decimal_points):
+def mk_vllm_json(prompt, num_beams):
+    return {
+        "prompt": prompt,
+        "n": 1,
+        "use_beam_search": num_beams > 1,
+        "best_of": num_beams,
+        "temperature": 0,
+        "stop": [";", "```"],
+        "max_tokens": 1024,
+    }
+
+
+def mk_tgi_json(prompt, num_beams):
+    # see swagger docs for /generate for the full list of parameters:
+    # https://huggingface.github.io/text-generation-inference/#/Text%20Generation%20Inference/generate
+    return {
+        "inputs": prompt,
+        "parameters": {
+            "best_of": num_beams,
+            "do_sample": num_beams > 1,
+            "return_full_text": False,
+            "max_new_tokens": 1024,
+        },
+    }
+
+
+def process_row(row, api_url: str, api_type: str, num_beams: int, decimal_points: int):
     start_time = time()
+    if api_type == "tgi":
+        json_data = mk_tgi_json(row["prompt"], num_beams)
+    elif api_type == "vllm":
+        json_data = mk_vllm_json(row["prompt"], num_beams)
+    else:
+        raise ValueError(f"Invalid api_type: {api_type}")
     r = requests.post(
         api_url,
-        json={
-            "prompt": row["prompt"],
-            "n": 1,
-            "use_beam_search": num_beams > 1,
-            "best_of": num_beams,
-            "temperature": 0,
-            "stop": [";", "```"],
-            "max_tokens": 4000,
-        },
+        json=json_data,
     )
     end_time = time()
     if "[SQL]" not in row["prompt"]:
         generated_query = (
             r.json()["text"][0].split("```")[-1].split("```")[0].split(";")[0].strip()
             + ";"
         )
+    elif api_type == "tgi":
```

Comment:
Hmm should this by higher up somewhere?

Reply:
good point, shifted it to the top of this if else block